### PR TITLE
fix: enable locks for updating caches in ModuleManager.

### PR DESCRIPTION
**Pull request by diafour**
#### Overview

ModuleManager component stores caches with values and patches. This PR fixes the race when these caches update from parallel queues.

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Review by nabokihms — COMMENTED** [review]
I looked through it briefly, and everything seemed ok. Let's wait for the @yalosev's review before merging it.

**Comment by diafour** [comment]
I've renamed a lock field and made it private.

**Review by yalosev — APPROVED** [review]

**Pull request merged**
